### DISCUSSION TITLE
DoubleTransition is not comptiable with attached properties, such as Canvas.Left and Canvas.Right

### DISCUSSION THREAD

**Discussion by wenyaoshao** [AvaloniaUI/Avalonia#16316] [Q&A] [answered]
I put a image in canvas and want to change Canvas.Left and Canvas.Right to move the image. 
But I got an error when using DoubleTransition
![Untitled picture](https://github.com/user-attachments/assets/bb8e3d4a-609b-4a2f-be9f-e9ea763449d5)

**Comment by IanRawley** [comment, marked as answer]
As it says, To isn't a property of a DoubleTransition.  You need to tell it what property to watch, not the value it should have.

Have a look at https://docs.avaloniaui.net/docs/guides/graphics-and-animation/transitions

**Reply by wenyaoshao** [reply]
It works now. 

        <Grid x:Name="tool" Canvas.Top="10" Background="Black">
          <Grid.Transitions>
            <Transitions>
              <DoubleTransition Property="Canvas.Left" Duration="0:0:0.2"/>
            </Transitions>
          </Grid.Transitions>
          <Grid.Styles>
            <Style Selector="Grid:pointerover">
              <Setter Property="Canvas.Left" Value="30"/>
            </Style>
            <Style Selector="Grid">
              <Setter Property="Canvas.Left" Value="10"/>
            </Style>
          </Grid.Styles>

But if I put Canvas.Left into 
<Grid x:Name="tool" Canvas.Top="10" **Canvas.Left="10"**  Background="Black">
It builds successfully but transtion can not be triggered.

**Reply by IanRawley** [reply]
I think you need to swap the order of the styles.  Generally speaking last Style wins, so more specific styles should be after general styles.
